### PR TITLE
Enable loading multiplayer with QMod installed

### PR DESCRIPTION
**Pull request by Coding-Hen**
Stop qmod initialisation on load and restore on exit

### CODE REVIEW

**Review by Jannify — COMMENTED** [review]
The solution seems fine to me. Would add info logging on qmod restore/remove and error logging if the file is not found.

**Comment by Coding-Hen** [comment]
> 
> 
> The solution seems fine to me. Would add info logging on qmod restore/remove and error logging if the file is not found.

Yes good idea will add that in. Will also remove the existing checks for qmod being installed so that our code is cleaned up. If the patching fails for whatever reason and they are still able to launch into the game qmods will still catch Nitrox being installed and throw an error.

**Review by Jannify — APPROVED** [review]

File: NitroxLauncher/Patching/QModHelper.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+﻿using System.IO;
+using NitroxModel.Logger;
+
+namespace NitroxLauncher.Patching
+{
+    public static class QModHelper
+    {
+        private static readonly string originalFileName = "winhttp.dll";
```
Comment (dartasen) [inline NitroxLauncher/Patching/QModHelper.cs:8]: minor: const

File: NitroxLauncher/Patching/QModHelper.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            string fileToRenamePath = Path.Combine(subnauticaBasePath, fileToRename);
+            if (!File.Exists(fileToRenamePath))
+            {
+                Log.Error($"QMod entry cannot be found, please uninstall QMod");
+                return;
+            }
+
+            QModPatched = !QModPatched;
+            string newFilePath = Path.Combine(subnauticaBasePath, newFileName);
+            File.Move(fileToRenamePath, newFilePath);
```
Comment (dartasen) [inline NitroxLauncher/Patching/QModHelper.cs:42]: Might be good to handle exceptions, especially with IO actions.

File: NitroxLauncher/Patching/QModHelper.cs (hunk 1 above)
Comment (Coding-Hen) [reply NitroxLauncher/Patching/QModHelper.cs:9]: Const cant be used here as static and readonly cant be used in conjunction with const

Comment (Measurity) [reply NitroxLauncher/Patching/QModHelper.cs:9]: A const is by default a static so you can remove both `static` and `readonly` to make this work.

Comment (Coding-Hen) [reply NitroxLauncher/Patching/QModHelper.cs:9]: Good to know, will make the change

**Pull request merged**